ledger/src/blockstore.rs

```diff
@@ -2015,41 +1993,96 @@ impl Blockstore {
                 }
             }
         };
+        get_before_slot_timer.stop();
 
         // Fetch the list of signatures that affect the given address
         let first_available_block = self.get_first_available_block()?;
-        let first_address_slot = self.get_lowest_slot_for_address(address)?;
-        if first_address_slot.is_none() {
-            return Ok(vec![]);
-        }
-        let lower_bound = cmp::max(first_available_block, first_address_slot.unwrap());
         let mut address_signatures = vec![];
-        loop {
-            if address_signatures.len() >= limit {
-                address_signatures.truncate(limit);
+
+        // Get signatures in `slot`
+        let mut get_initial_slot_timer = Measure::start("get_initial_slot_timer");
+        let mut signatures = self.find_address_signatures(address, slot, slot)?;
+        signatures.reverse();
+        if let Some(excluded_signatures) = excluded_signatures.take() {
+            address_signatures.extend(
+                signatures
+                    .into_iter()
+                    .filter(|(_, signature)| !excluded_signatures.contains(&signature)),
+            )
+        } else {
+            address_signatures.append(&mut signatures);
+        }
+        get_initial_slot_timer.stop();
+
+        // Check active_transaction_status_index to see if it contains slot. If so, start with that
```

Comment:
Would you mind adding a comment somewhere about what `active_transaction_status_index` is used for, for future code readers.  It's a little unclear to me, "a thing that is 0 or 1"

Reply:
I added several docs/comments. Would appreciate if you could see if they make sense.